### PR TITLE
Plugin/projects: cleanup

### PR DESCRIPTION
**Pull request by gaelicWizard**
## Description
`shfmt`, `shellcheck`, and minor cleanup.

## Motivation and Context
Just cleaning up.

## Types of changes
- [x] Cleanup (non-breaking change which fixes an issue)

## Checklist:
- [x] My code follows the code style of this project.
- [x] If my change requires a change to the documentation, I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] If I have added a new file, I also added it to ``clean_files.txt`` and formatted it using ``lint_clean_files.sh``.
- [ ] I have added tests to cover my changes, and all the new and existing tests pass.

### CODE REVIEW

File: plugins/available/projects.plugin.bash [hunk 1]
```diff
@@ -1,75 +1,51 @@
...
-  return 1 # should never reach this
-
-fi
-
-
-$cmd "$dest"
+# shellcheck shell=bash
+about-plugin 'quickly navigate configured project paths'
+
+: "${PROJECT_PATHS:=$HOME/Projects:$HOME/src:$HOME/work}"
```
Comment (NoahGorny) [inline plugins/available/projects.plugin.bash:4]: I think we should rename this parameter to `BASH_IT_PROJECT_PATHS` and encourage users to define it themselves in the bashrc template

Comment (gaelicWizard) [reply plugins/available/projects.plugin.bash:4]: Agreed! I'll change it when I'm back at my computer.

Comment (NoahGorny) [reply plugins/available/projects.plugin.bash]: We should add a log about it if the paths do not exist, to let the users know that such variable exist

Comment (gaelicWizard) [reply plugins/available/projects.plugin.bash]: The paths almost certainly won't exist in most cases. I don't have a `~/src` or `~/work` directory

Comment (gaelicWizard) [reply plugins/available/projects.plugin.bash]: I've added an example to the template

File: template/bash_profile.template.bash [hunk 2]
```diff
@@ -34,6 +34,9 @@ export IRC_CLIENT='irssi'
 # Set this to the command you use for todo.txt-cli
 export TODO="t"
 
+# Set this to the location of your work or project folders
+#BASH_IT_PROJECT_PATHS="${HOME}/Projects:/Volumes/work/src"
```
Comment (NoahGorny) [inline template/bash_profile.template.bash:38]: you forgot the `export` 😄

Comment (gaelicWizard) [reply template/bash_profile.template.bash:38]: Nope! Almost none of these should be exported, but I didn't fix the others here. 😃

Comment (NoahGorny) [reply template/bash_profile.template.bash:38]: you are absolutely correct

File: plugins/available/projects.plugin.bash (hunk 1 above)
Comment (gaelicWizard) [reply plugins/available/projects.plugin.bash]: I ended up refactoring and made an easy spot for an error message

**Comment by gaelicWizard** [comment]
I rebased on current master and refactored a bit. I think this one should be more reliable, and should help the user figure it out if needed.

**Review by NoahGorny — APPROVED** [review]

**Pull request merged**